app/src/main/java/zapsolutions/zap/walletManagement/WalletDetailsActivity.java

```diff
@@ -0,0 +1,208 @@
+
+            Button changeBtn = findViewById(R.id.buttonChangeConnection);
+            changeBtn.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Intent intent = new Intent(WalletDetailsActivity.this, ConnectRemoteNodeActivity.class);
+                    intent.putExtra(ManageWalletsActivity.WALLET_ID, mId);
+                    startActivity(intent);
+                }
+            });
+        }
+
+        Button switchBtn = findViewById(R.id.buttonActivate);
+        switchBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                PrefsUtil.edit().putString(PrefsUtil.CURRENT_WALLET_CONFIG, mId).commit();
+
+                // Do not ask for pin again...
+                TimeOutUtil.getInstance().restartTimer();
+
+                Wallet.getInstance().reset();
+
+                if (!getWalletConfig().isLocal()) {
+                    // ToDo: Stop local LND if we switch to a remote wallet (right now it stops, but then it crashes when switching back to Local LND
+                    openHome();
+                } else {
+                    openHome();
+                }
+            }
+        });
+
+        Button renameBtn = findViewById(R.id.buttonRename);
+        renameBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                showWalletNameInput();
+            }
+        });
+
+        Button deleteBtn = findViewById(R.id.buttonDelete);
+        deleteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (PrefsUtil.getCurrentWalletConfig().equals(mId)) {
+                    Toast.makeText(WalletDetailsActivity.this, "You cannot delete the currently active wallet.", Toast.LENGTH_LONG).show();
+                } else {
+                    deleteWallet();
+                }
+            }
+        });
+    }
+
+    private void showWalletNameInput() {
+        AlertDialog.Builder adb = new AlertDialog.Builder(this);
+        adb.setTitle(R.string.wallet_name);
+        adb.setCancelable(false);
+        View viewInflated = LayoutInflater.from(this).inflate(R.layout.dialog_input_text, null, false);
+
+        final EditText input = viewInflated.findViewById(R.id.input);
+        input.setShowSoftInputOnFocus(true);
+        input.setText(getWalletConfig().getAlias());
+        input.requestFocus();
+
+        mInputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
+
+        adb.setView(viewInflated);
+
+        adb.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                if (input.getText().toString().isEmpty()) {
```

Comment:
We could add `.trim()` here, would not allow wallets that only contain spaces/tabs, and get rid of those at start/end.